storage/src/main/java/org/apache/kafka/storage/internals/log/PendingPartitionChange.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.storage.internals.log;
+
+import org.apache.kafka.metadata.LeaderAndIsr;
+
+public interface PendingPartitionChange extends PartitionState {
+    CommittedPartitionState lastCommittedState();
+    LeaderAndIsr sentLeaderAndIsr();
+    void notifyListener(AlterPartitionListener alterPartitionListener);
```

Comment:
`notifyListener` is used for a single purpose. Perhaps we can decouple `AlterPartitionListener` from `PendingPartitionChange`. for example:
```scala
      proposedIsrState match {
        case _: PendingExpandIsr => alterPartitionListener.markIsrExpand()
        case _: PendingShrinkIsr => alterPartitionListener.markIsrShrink()
      }
```